gattc_darwin.go

```diff
@@ -15,13 +15,7 @@ import (
 // Passing a nil slice of UUIDs will return a complete list of
 // services.
 func (d *Device) DiscoverServices(uuids []UUID) ([]DeviceService, error) {
-	cbuuids := []cbgo.UUID{}
-	for _, u := range uuids {
-		uuid, _ := cbgo.ParseUUID(u.String())
-		cbuuids = append(cbuuids, uuid)
-	}
-
-	d.prph.DiscoverServices(cbuuids)
+	d.prph.DiscoverServices([]cbgo.UUID{})
```

Comment:
Why are you no longer passing the `uuids` slice to `DiscoverServices`? See the [documentation](https://developer.apple.com/documentation/corebluetooth/cbperipheral/1518706-discoverservices):

> **Note**
>
> If the `servicesUUIDs` parameter is `nil`, this method returns all of the peripheral’s available services. This is much slower than providing an array of service UUIDs to search for.

Reply:
It is not actually working, it seems. Not sure if that is because of cbgo or something else. I tried a whole bunch of different things but could not get it to work as documented.

Reply:
That is unfortunate. What if you put in a literal `nil` instead of `[]cbgo.UUID{}`? (That's in fact semantically different, `[]cbgo.UUID{}` is not equal to `nil`).

Reply:
Looking at the source of cbgo that is unlikely to have an effect. I'm working on a possible patch to cbgo.

Reply:
Never mind, I'm confused.
This is indeed rather unfortunate. Maybe you can raise a bug at the cbgo repository?